client/components/redux-forms/redux-form-radio/index.jsx

```diff
@@ -1,43 +1,24 @@
+/** @format */
 /**
  * External dependencies
  */
-import React, { Component, PropTypes } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
 import { Field } from 'redux-form';
-import { omit } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import FormRadio from 'components/forms/form-radio';
 
-class ReduxFormRadio extends Component {
-	static propTypes = {
-		name: PropTypes.string,
-		value: PropTypes.string,
-	};
+// eslint-disable-next-line no-unused-vars
+const RenderRadio = ( { input, meta, type, ...props } ) => <FormRadio { ...input } { ...props } />;
```

Comment:
Thanks for looking at this @ockham!

> Components should be nouns, not verbs (`Render...`) -- should we just call it `Radio` now?

Yes, I'm finding it challenging to come up with good names for these components. We need to distinguish between three kinds of components that are closely related:
- `FormRadio` -- the enhanced equivalent of plain `<input type="radio">`
- `RenderFormRadio` -- maps props between what `Field` passes down (`{ input, meta, ...props }`) and what the `<input type="radio">` expects (`value`, `onChange`)
- `ReduxFormRadio` -- wrapper around `Field`, essentially a `connect()`-ed `<input>`

Maybe `Radio` is sufficient, although not very distinctive. Or `FormRadioRenderer`? Ideas welcome.

> Also, should we continue to destructure `input` so it's more explicit?

The `input` object contains additional handlers, like `onFocus`, `onBlur`, `onDragStart`, `onDrop`. Redux Form maintains a `touched` flag for every field, set to `true` after it's focused and blurred. Useful for smart validation.

I think it's safer to just destructure the whole thing. It makes sure that all props are passed down and nothing is forgotten.